identity/aziot-identityd/src/error.rs

```diff
@@ -21,6 +21,25 @@ impl Error {
     {
         Error::InvalidParameter(name, err.into())
     }
+
+    pub(crate) fn is_network(&self) -> bool {
+        match &self {
+            Error::DpsClient(err) | Error::HubClient(err) => match err.kind() {
+                std::io::ErrorKind::TimedOut => true, // Network timed out
+                std::io::ErrorKind::Other => {
+                    if let Some(err) = err.get_ref() {
+                        // TODO: check if error is hyper::Error
```

Comment:
@arsing Rust noob question: How do I check that err is a hyper::Err here? err.get_ref() just gives something that impl std::err::Error. I want to do something like:

```
if let Some(err) = err.get_ref() {
    if err is hyper::Error {
        if err.is_connect() {
            return true;
        }
    }
}
```

Reply:
[`.downcast_ref()`](https://doc.rust-lang.org/stable/std/error/trait.Error.html#method.downcast_ref) but it's an anti-pattern. When you want to look at inner errors, you *should* make the outer error hold the inner error strongly-typed. In this case that would mean replacing `std::io::Error` with `enum Error { Http(hyper::Error), ... }`

Reply:
Got it. Actually I figured we can just map hyper's `err.is_connected()` to `std::io::ErrorKind::NotConnected`, so I did that.